framework/src/security/security_utils.c

```diff
@@ -338,14 +324,17 @@ int secutils_convert_ecdsaparam_s2h(security_ecdsa_param *eparam, hal_ecdsa_mode
 
 int secutils_convert_dhparam_s2h(security_dh_param *dparam, hal_dh_data *hdata)
 {
-	if (!dparam || !hdata) {
+	if (!hdata) {
 		return -1;
 	}
 
-	if (!dparam->G || !dparam->P || !dparam->pubkey) {
+	if (!dparam || !dparam->G || !dparam->P || !dparam->pubkey) {
 		return -1;
 	}
 	hdata->mode = secutils_convert_dhmode_s2h(dparam->mode);
+	if (hdata->mode == HAL_DH_UNKNOWN) {
+		return -1;
```

Comment:
please make a macro instead of hard code value (-1).

Reply:
it's an internal function so I think hard code value is ok.

Reply:
ok.